### PR TITLE
Allow setting `readOnly` on `FileSelect`

### PR DESCRIPTION
**Pull request by jamesricky**
---

## PR Checklist

-   [x] Merge dependent PR https://github.com/vivid-planet/comet/pull/2333
-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-801
-   [x] Provide screenshots/screencasts if the change contains visual changes

Screenshot 2024-07-24 at 12 02 47

### CODE REVIEW

File: packages/admin/admin/src/form/file/FileSelect.tsx [hunk 1]
```diff
@@ -134,7 +154,7 @@ export const FileSelect = <AdditionalValidFileValues = Record<string, unknown>,>
                     </ErrorMessage>
                 </Error>
             )}
-            {Boolean(filesInfoText) && <FilesInfoText {...slotProps?.filesInfoText}>{filesInfoText}</FilesInfoText>}
+            {Boolean(filesInfoText && !readOnly) && <FilesInfoText {...slotProps?.filesInfoText}>{filesInfoText}</FilesInfoText>}
```
Comment (johnnyomair) [inline packages/admin/admin/src/form/file/FileSelect.tsx:157]: Why shouln't it not be shown when the select is read-only?

File: packages/admin/admin/src/form/file/fileSelectItemTypes.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { ReactNode } from "react";
 
 export type ValidFileSelectItem<AdditionalFileValues = Record<string, unknown>> = {
     name: string;
-    size: number;
+    size?: number;
```
Comment (johnnyomair) [inline packages/admin/admin/src/form/file/fileSelectItemTypes.ts:5]: Why can this be optional?

**Review by johnnyomair — COMMENTED** [review]
Please rebase, thanks!

File: packages/admin/admin/src/form/file/FileSelect.tsx (hunk 1 above)
Comment (jamesricky) [reply packages/admin/admin/src/form/file/FileSelect.tsx:157]: This is only information about what files can be uploaded, that is irrelevant when uploading is not possible (which is the case with `readOnly`)

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**